### PR TITLE
feat(bull-mq): adds support for filter config opt

### PR DESCRIPTION
**Pull request by pabloerhard**
### What does this PR do?

Adds a filter option to the BullMQ plugin so users can decide which producer jobs should be instrumented.

The filter is applied to Queue.add, Queue.addBulk, and FlowProducer.add. Rejected jobs skip span creation, trace context injection, and DSM checkpoint handling.

This addresses #7641

### CODE REVIEW

**Comment by watson** [comment]
A [new linting rule](https://github.com/DataDog/dd-trace-js/pull/8537) just landed in `master` that will fail this PR if merged as-is. I'll merge in `master` so that it's more clear - sorry for the inconvenience.

File: packages/datadog-plugin-bullmq/src/producer.js [hunk 1]
```diff
@@ -87,6 +112,15 @@ class BaseBullmqProducerPlugin extends ProducerPlugin {
 class QueueAddPlugin extends BaseBullmqProducerPlugin {
   static prefix = 'tracing:orchestrion:bullmq:Queue_add'
 
+  shouldInstrument (ctx) {
+    return this.config.producerFilter({
```
Comment (BridgeAR) [inline packages/datadog-plugin-bullmq/src/producer.js:118]: This is an interesting case: the default producerFilter is a noop method. We always create an object for each call though. So instead, we could check against that noop as a reference check and create nothing new. That should safe the allocation for the default case.

The check should probably be in `bindStart` to only do all of this when `this.config.producerFilter !== defaultFilter`.

File: packages/datadog-plugin-bullmq/src/filter.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+/**
+ * Resolve a user-provided filter from plugin config. If the value is present
+ * but not a function, log an error and fall back to the default allow-all
+ * filter. Mirrors the contract used by `plugins/util/urlfilter#getFilter`.
+ *
+ * @param {{ producerFilter?: unknown }} config Plugin config that may carry a `producerFilter`.
+ * @returns {BullmqFilter}
+ */
+function getFilter (config) {
```
Comment (BridgeAR) [inline packages/datadog-plugin-bullmq/src/filter.js:26]: Could we unify that somehow with the referenced method? Having these in different spots makes it difficult to prevent drifting functionality (while it does seem to at least partially be intentional).

Comment (pabloerhard) [reply packages/datadog-plugin-bullmq/src/filter.js:26]: Yeah this was intentional considering these are not all identical, I'll look further on this coul dbe extracted into a more general function.

**Review by BridgeAR — COMMENTED** [review]
The FlowProducer.add method would still need a test case for the coverage minimum :)

File: packages/datadog-plugin-bullmq/src/producer.js [hunk 3]
```diff
@@ -128,25 +164,59 @@ class QueueAddPlugin extends BaseBullmqProducerPlugin {
...
   getSpanData (ctx) {
     const queueName = ctx.self?.name || 'bullmq'
-    const jobs = ctx.arguments?.[0]
+    const jobs = this.#getFilteredJobs(ctx)
     return {
       resource: queueName,
       meta: {
         'messaging.destination.name': ctx.self?.name,
-        'messaging.batch.message_count': Array.isArray(jobs) ? jobs.length : undefined,
+        'messaging.batch.message_count': jobs?.length,
```
Comment (BridgeAR) [inline packages/datadog-plugin-bullmq/src/producer.js:212]: Is this change correct / intentional? The number of jobs queued is still the same as before, no matter that some were filtered here

Comment (pabloerhard) [reply packages/datadog-plugin-bullmq/src/producer.js:212]: True, thank you for catching that! I guess we could ask ourselves the same for DSM? Do we want to limit this to only tracing or tracing + dsm?

Comment (BridgeAR) [reply packages/datadog-plugin-bullmq/src/producer.js:212]: I think this is already working as expected

**Review by BridgeAR — APPROVED** [review]
LGTM
